data race occurs in the test

```
$ go version
go version go1.14.4 linux/amd64
$ git checkout master
$ go test ./... -race
==================
WARNING: DATA RACE
Write at 0x00c0002a00a0 by goroutine 71:
  github.com/Songmu/flextime.(*fakeTimer).send()
      /home/heijo/ghq/github.com/Songmu/flextime/faketimer.go:53 +0x6a
  github.com/Songmu/flextime.(*fakeTimer).Reset()
      /home/heijo/ghq/github.com/Songmu/flextime/faketimer.go:94 +0x280
  github.com/Songmu/flextime.newFakeTicker.func1()
      /home/heijo/ghq/github.com/Songmu/flextime/faketicker.go:33 +0xf1

Previous read at 0x00c0002a00a0 by goroutine 68:
  github.com/Songmu/flextime.(*fakeTimer).isActive()
      /home/heijo/ghq/github.com/Songmu/flextime/faketimer.go:35 +0x47
  github.com/Songmu/flextime.(*fakeTimer).Stop()
      /home/heijo/ghq/github.com/Songmu/flextime/faketimer.go:99 +0x3c
  github.com/Songmu/flextime.(*fakeTicker).Stop()
      /home/heijo/ghq/github.com/Songmu/flextime/faketicker.go:47 +0x50
  github.com/Songmu/flextime.(*Ticker).Stop()
      /home/heijo/ghq/github.com/Songmu/flextime/clock.go:65 +0x211
  github.com/Songmu/flextime_test.runTests.func9()
      /home/heijo/ghq/github.com/Songmu/flextime/func_test.go:151 +0x1e8
  testing.tRunner()
      /home/linuxbrew/.linuxbrew/Cellar/go/1.14.4/libexec/src/testing/testing.go:991 +0x1eb

Goroutine 71 (running) created at:
  github.com/Songmu/flextime.newFakeTicker()
      /home/heijo/ghq/github.com/Songmu/flextime/faketicker.go:27 +0x16f
  github.com/Songmu/flextime.(*fakeClock).NewTicker()
      /home/heijo/ghq/github.com/Songmu/flextime/fakeclock.go:51 +0xac
  github.com/Songmu/flextime.NewTicker()
      /home/heijo/ghq/github.com/Songmu/flextime/flextime.go:74 +0x4c
  github.com/Songmu/flextime_test.runTests.func9()
      /home/heijo/ghq/github.com/Songmu/flextime/func_test.go:148 +0x65
  testing.tRunner()
      /home/linuxbrew/.linuxbrew/Cellar/go/1.14.4/libexec/src/testing/testing.go:991 +0x1eb

Goroutine 68 (finished) created at:
  testing.(*T).Run()
      /home/linuxbrew/.linuxbrew/Cellar/go/1.14.4/libexec/src/testing/testing.go:1042 +0x660
  github.com/Songmu/flextime_test.runTests()
      /home/heijo/ghq/github.com/Songmu/flextime/func_test.go:146 +0x7d2
  github.com/Songmu/flextime_test.TestFix()
      /home/heijo/ghq/github.com/Songmu/flextime/func_test.go:178 +0x44
  testing.tRunner()
      /home/linuxbrew/.linuxbrew/Cellar/go/1.14.4/libexec/src/testing/testing.go:991 +0x1eb
==================
--- FAIL: TestSet (0.03s)
    testing.go:906: race detected during execution of test
FAIL
exit status 1
FAIL    github.com/Songmu/flextime      3.398s
```

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ devel-deps:
 
 .PHONY: test
 test:
-	go test
+	go test -race
 
 .PHONY: lint
 lint: devel-deps

--- a/faketimer.go
+++ b/faketimer.go
@@ -26,8 +26,10 @@ func newFakeTimer(c NowSleeper, d time.Duration, f func()) *fakeTimer {
 		ch:   make(chan time.Time, 1),
 		inch: make(chan time.Time),
 		stop: make(chan struct{}, 1),
+		done: make(chan struct{}),
 		fun:  f,
 	}
+	close(fti.done)
 	fti.Reset(d)
 	return fti
 }
@@ -46,12 +48,8 @@ func (fti *fakeTimer) renewDone() chan struct{} {
 }
 
 func (fti *fakeTimer) isActive() bool {
-	done := fti.doneCh()
-	if done == nil {
-		return false
-	}
 	select {
-	case <-done:
+	case <-fti.doneCh():
 		return false
 	default:
 		return true

--- a/faketimer.go
+++ b/faketimer.go
@@ -14,6 +14,7 @@ type fakeTimer struct {
 
 	ch, inch   chan time.Time
 	stop, done chan struct{}
+	doneMu     sync.RWMutex
 	triggerAt  time.Time
 }
 
@@ -31,12 +32,26 @@ func newFakeTimer(c NowSleeper, d time.Duration, f func()) *fakeTimer {
 	return fti
 }
 
+func (fti *fakeTimer) doneCh() chan struct{} {
+	fti.doneMu.RLock()
+	defer fti.doneMu.RUnlock()
+	return fti.done
+}
+
+func (fti *fakeTimer) renewDone() chan struct{} {
+	fti.doneMu.Lock()
+	defer fti.doneMu.Unlock()
+	fti.done = make(chan struct{})
+	return fti.done
+}
+
 func (fti *fakeTimer) isActive() bool {
-	if fti.done == nil {
+	done := fti.doneCh()
+	if done == nil {
 		return false
 	}
 	select {
-	case <-fti.done:
+	case <-done:
 		return false
 	default:
 		return true
@@ -50,7 +65,7 @@ func (fti *fakeTimer) C() <-chan time.Time {
 // The `send` is called only inside `Reset` and exclusive control is performed on the `Reset` side,
 // so the `send` itself need not do exclusive control.
 func (fti *fakeTimer) send() {
-	fti.done = make(chan struct{})
+	done := fti.renewDone()
 
 	go func() {
 		select {
@@ -60,7 +75,7 @@ func (fti *fakeTimer) send() {
 			} else {
 				fti.ch <- t
 			}
-		case <-fti.done:
+		case <-done:
 		}
 	}()
 
@@ -72,7 +87,7 @@ func (fti *fakeTimer) send() {
 		}():
 		case <-fti.stop:
 		}
-		close(fti.done)
+		close(done)
 	}()
 }
 
@@ -102,7 +117,7 @@ func (fti *fakeTimer) Stop() bool {
 	// behaves like that.
 	if active {
 		fti.stop <- struct{}{}
-		<-fti.done
+		<-fti.doneCh()
 		// The Timer may be fired at the same timing as the Stop. Also, the multiple `Reset` may be called
 		// concurrently. In that case, `struct{}{}` could be accumulated in the channel, so drain it here.
 		select {

